src/backend/distributed/commands/extension.c

```diff
@@ -491,6 +495,153 @@ PreprocessAlterExtensionUpdateStmt(Node *node, const char *queryString)
 }
 
 
+/*
+ * PostprocessAlterExtensionCitusUpdateStmt is called after ALTER EXTENSION
+ * citus UPDATE command is executed by standard utility hook.
+ *
+ * Actually, we do not need such a post process function for ALTER EXTENSION
+ * UPDATE commands unless the extension is Citus itself. This is because we
+ * need to mark existing objects that are not included in distributed object
+ * infrastructure in older versions of Citus (but now should be) as distributed
+ * if we really updated Citus to the available version successfully via standard
+ * utility hook.
+ */
+void
+PostprocessAlterExtensionCitusUpdateStmt(Node *node, bool
+										 citusCanBeUpdatedToAvailableVersion)
+{
+	/*
+	 * We should not postprocess this command in workers as they do not keep track
+	 * of citus.pg_dist_object.
+	 */
+	if (!IsCoordinator())
+	{
+		return;
+	}
+
+	/*
+	 * If Citus version was already same with available version before standard
+	 * process utility runs ALTER EXTENSION command, then Citus is not updated
+	 * actually, no need to post process
+	 */
+	if (!citusCanBeUpdatedToAvailableVersion)
+	{
+		return;
+	}
+
+	bool citusIsUpdatedToLatestVersion = InstalledAndAvailableVersionsSame();
+
+	/*
+	 * Knowing that Citus version was different than the available version before
+	 * standard process utility runs ALTER EXTENSION command, we perform post
+	 * process operations if Citus is updated to that available version
+	 */
+	if (!citusIsUpdatedToLatestVersion)
+	{
+		return;
+	}
+
+	/* *check if pg_dist_object table exists */
+	Oid pg_dist_object_oid = get_relname_relid("pg_dist_object",
+											   CitusCatalogNamespaceId());
+	bool distObjectRelationExists = OidIsValid(pg_dist_object_oid);
+
+	/*
+	 * If pg_dist_object table does not exist, then we cannot introduce existing
+	 * objects to the distributed object infrastructure
+	 */
+	if (!distObjectRelationExists)
+	{
```

Comment:
Can this ever happen after checking `InstalledAndAvailableVersionsSame()`? 

Presumably, we either installed an older version of Citus, in which case we wouldn't be running this code, or the SQL still lags behind the installed version, in which case we would have already returned.

Reply:
that exactly makes sense, removing this check